src/js/components/CourseModalComponent.js

```diff
@@ -75,8 +69,11 @@ module.exports = React.createBackboneClass({
     this.state.course[e.currentTarget.getAttribute('id')] = e.currentTarget.value;
   },
 
-  changeTextbookValue(e) {
-    this.state.course.textbook = this.props.textbooks.get(e.currentTarget.value);
+  selectTextbooks(options) {
+    this.setState({ textbooks: options });
+    this.state.course.textbooks = options.map(textbook => {
```

Comment:
Shouldn't be setting state directly, only by using `this.setState`.

Reply:
Yes, always set with a new value

```js
this.setState({
  textbooks: options,
  course: {
    ...this.state.course,
    textbooks: options.map(textbook => textbook.value)
  }  
})
```

Reply:
Yes, always set with a new value

```js
this.setState({
  textbooks: options,
  course: {
    ...this.state.course,
    textbooks: options.map(textbook => textbook.value)
  }  
})
```